target/linux/ramips/dts/WN-AC733GR3.dts

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "iodata,wn-ac733gr3", "ralink,mt7620a-soc";
+	model = "I-O DATA WN-AC733GR3";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "wn-ac733gr3:green:power";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		notification {
+			label = "wn-ac733gr3:green:notification";
+			gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "wn-ac733gr3:green:wlan2g";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "wn-ac733gr3:green:wlan5g";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		router {
+			label = "router";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+
+		auto {
+			label = "auto";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+	};
+
+	regulator {
```

Comment:
Is this really a power line regulator for the rtl8367rb? Or is this regulator here just so it can be used to control the rtl8367rb's reset line?

Because if this is just the reset line, then we are in a bit of a pickle. On one hand, you could do away with the regulator here (so you don't need to add the symbol in the kernel config) and go straight for a gpio-hog instead. 

Or on the other hand do a bit of preparation. Because the upstream linux kernel driver will do this differently, since it has a "reset-gpios" property that fits perfectly: https://www.kernel.org/doc/Documentation/devicetree/bindings/net/dsa/realtek-smi.txt

Reply:
I think that it is probably a reset line. (I cannot be confident because I cannot find the document for RTL8367RB...)
OK, I'll try to use gpio-hog.

Implementation in Linux Kernel, I have been interested for a while. However, I do not know how to configure the interrupt controller correctly.

Reply:
>I think that it is probably a reset line. (I cannot be confident because I cannot find the document for RTL8367RB...)

Well, If you want to measure it to confirm whenever this is the reset line or not: According to my RTL8367RB.pdf (found by google, same chip is used in the WNDAP620/660 and a few other) the nRESET signal is supposed to be on Pin 91 of the LQFP-128 package. This would be the 6th pin from the top left corner of the chip. if said chip is oriented in such a way that the Pin 1 marker is in the bottom/lower-left corner (i.e.: the text and the realtek crab has the "proper" orientation).

Reply:
Thank you for the information, I'll check it.

Reply:
> Thank you for the information, I'll check it.

Forgot to add the datasheets comment on the 91 pin: "When low active will reset the RTL8367RB. ".

As for power pins: there are several analog and digital voltages ranging from 3.3, 2.5 and 1.0 volts.

Reply:
Thank you, I'll keep in mind.

Reply:
result (GPIO 0):
- set LOW  -> RTL8367RB RESET: 0 V
- set HIGH -> RTL8367RB RESET: 3.3V